tuna/driver.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+###############################################################################
+#
+# MIT License
+#
+# Copyright (c) 2023 Advanced Micro Devices, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+###############################################################################
+"""Module that encapsulates the DB representation of a Driver cmd"""
+
+from typing import Union, Dict, Any
+from abc import ABC, abstractmethod
+from tuna.utils.logger import setup_logger
+
+LOGGER = setup_logger('driver_base')
+
+
+class DriverBase(ABC):
+  """Represents db tables based on ConfigType"""
+
+  @abstractmethod
+  def construct_driver(self, line: str) -> bool:
+    """Takes a MIOpenDriver cmd or PDB key"""
+    raise NotImplementedError("Not implemented")
+
+  @abstractmethod
+  def construct_driver_from_db(self, db_obj: Any) -> bool:
+    """Takes a <>_config row and returns a driver cmd"""
+    raise NotImplementedError("Not implemented")
+
+  @abstractmethod
+  def insert_tensor(self, tensor_dict: dict) -> int:
```

Comment:
I'm anticipating that tensors won't be generally applicable for most use cases.

Reply:
ditto! move all the tensor stuff to miopen, insert, compose, get_input_t,...all that.